helpdesk_mgmt_timesheet/models/hr_timesheet.py

```diff
@@ -25,5 +25,7 @@ class AccountAnalyticLine(models.Model):
     @api.onchange("ticket_id")
     def onchange_ticket_id(self):
         for record in self:
+            if not record.ticket_id:
+                continue
```

Comment:
```suggestion
        for record in self.filtered("ticket_id"):
```

Reply:
@pedrobaeza I consider running a filtered in an onchange an antipattern for me on small recordsets, especially for what is normally a single record. Even with non newId record it is dog slow.

```
t = env['account.analytic.line'].search([], limit=1)
>>> timeit(stmt='bool(t.ticket_id)',globals=globals())
5.048047009004222
>>> timeit(stmt='t.filtered("ticket_id")',globals=globals())
19.299567326001124
>>> timeit(stmt='for x in t:\n' '  if not x.ticket_id:\n' '    continue\n' '  pass\n',globals=globals())
6.696332884996082

>>> t = self.env['account.analytic.line'].new()
>>> timeit(stmt='for x in t:\n' '  if not x.ticket_id:\n' '    continue\n' '  pass\n',globals=globals())
7.403129160004028
>>> timeit(stmt='for x in t.filtered("ticket_id"):\n'  '  pass',globals=globals())
21.471524859000056
```

Reply:
OK, it's weird that `filtered` loses such performance...

Reply:
Curious to know, in which case `filtered` give better performance?

Reply:
I expect similar performances. It's only syntactic sugar.

Reply:
I guess it makes sense because with `filtered()` you must:

1. Iterate over *all* records always.
1. Produce a new recordset. I noticed it calls `browse()` without `with_prefetch`, which could mean that new queries have to be executed. I'm not sure how this works after v13 ORM refactor though.
1. Iterate again over all records in the new recordset.

I tend to avoid it these days too. Where it will be less performant the ~100% of cases is when you can do a `break` instead of a `continue`.

Reply:
BTW I think https://github.com/OCA/helpdesk/pull/172#discussion_r508868961 should be reported as an issue to Odoo upstream.